ecs-init/docker/dependencies.go

```diff
@@ -32,22 +36,43 @@ type dockerclient interface {
 	StartContainer(id string, hostConfig *godocker.HostConfig) error
 	WaitContainer(id string) (int, error)
 	StopContainer(id string, timeout uint) error
+	Ping() error
 }
 
 type _dockerclient struct {
-	docker *godocker.Client
+	docker dockerclient
 }
 
-func newDockerClient() (*_dockerclient, error) {
+type dockerClientFactory interface {
+	NewVersionedClient(endpoint string, apiVersionString string) (dockerclient, error)
+}
+
+type godockerClientFactory struct{}
+
+func (client godockerClientFactory) NewVersionedClient(endpoint string, apiVersionString string) (dockerclient, error) {
+	return godocker.NewVersionedClient(endpoint, apiVersionString)
+}
+
+func newDockerClient(dockerClientFactory dockerClientFactory, pingBackoff backoff.Backoff) (dockerclient, error) {
 	dockerUnixSocketSourcePath, fromEnv := config.DockerUnixSocket()
 	if !fromEnv {
 		dockerUnixSocketSourcePath = "/var/run/docker.sock"
 	}
-	client, err := godocker.NewVersionedClient(config.UnixSocketPrefix+dockerUnixSocketSourcePath, "1.15")
+	client, err := dockerClientFactory.NewVersionedClient(
```

Comment:
At some point (does not need to be now) we should rip out this code.  go-dockerclient has a `NewVersionedClientFromEnv` that handles this properly.